src/app/new-auth/register/register.component.ts

```diff
@@ -1,15 +1,327 @@
+  animations: [
+    trigger('slideInRight', [
+      transition('* => *', [
+        style({ transform: 'translateX(100%)', opacity: 0 }),
+        animate('400ms', style({ transform: 'translateX(0)', opacity: 1 }))
+      ]),
+      transition('* => *', [
+        style({ transform: 'translateX(0)', opacity: 1 }),
+        animate('400ms', style({ transform: 'translateX(-100%)', opacity: 0 }))
+      ])
+    ]),
+
+    trigger('fallFromTop', [
+      transition('* => *', [
+        query(':enter', style({ opacity: 0 }), { optional: true }),
+
+        query(
+          ':enter',
+          stagger('300ms', [
+            animate(
+              '.6s ease-in',
+              keyframes([
+                style({ opacity: 0, transform: 'translateY(-75%)', offset: 0 }),
+                style({
+                  opacity: 0.5,
+                  transform: 'translateY(35px)',
+                  offset: 0.3
+                }),
+                style({ opacity: 1, transform: 'translateY(0)', offset: 1.0 })
+              ])
+            )
+          ]),
+          { optional: true }
+        )
+      ])
+    ])
+  ]
 })
-export class RegisterComponent implements OnInit {
+export class RegisterComponent implements OnInit, OnDestroy{
+  private ngUnsubscribe = new Subject<void>();
+  ssoRedirect = environment.apiURL + '/google';
+
+  TEMPLATES = {
+    info: { temp: 'info', index: 1 },
+    account: { temp: 'account', index: 2 },
+    submission: { temp: 'submission', index: 3 },
+    sso: { temp: 'sso', index: 3 }
+  };
+
+  currentTemp: String = 'info';
+  currentIndex = 1;
+
+  registrationFailure: Boolean = false;
+  verified = false;
+  siteKey = '6LfS5kwUAAAAAIN69dqY5eHzFlWsK40jiTV4ULCV';
+  errorMsg = 'There was an issue on our end with your registration, we are sorry for the inconvience.\n Please try again later!';
+  fieldErrorMsg = '';
+
+  slide: boolean;
+  fall = false;
+
+  regInfo = {
+    firstname: '',
+    lastname: '',
+    email: '',
+    organization: '',
+    password: '',
+    confirmPassword: '',
+    username: ''
+  };
+
+  buttonGuards = {
+    isInfoPageInvalid: true,
+    isRegisterPageInvalid: true
+  };
+
+  infoFormGroup: FormGroup = new FormGroup({
+    firstname: this.authValidation.getInputFormControl('required'),
+    lastname: this.authValidation.getInputFormControl('required'),
+    email: this.authValidation.getInputFormControl('email'),
+    organization: this.authValidation.getInputFormControl('required'),
+  });
+  accountFormGroup: FormGroup = new FormGroup({
+    username: this.authValidation.getInputFormControl('username'),
+    password: this.authValidation.getInputFormControl('password'),
+    confirmPassword: this.authValidation.getInputFormControl('required'),
+    captcha: new FormControl()
+  }, MatchValidator.mustMatch('password', 'confirmPassword'));
 
-  constructor() { }
+  emailInUse = false;
+  usernameInUse = false;
+  loading = false;
+
+  constructor(
```

Comment:
have you considered adding a redirect url so that if someone comes to clark for the first time, finds an object they like then registers, it takes them back to the LO they were looking at?